### ISSUE TITLE
[Components and pattern standards] Design components or patterns don't align with Design System guidelines. (04.07.5)

### ISSUE THREAD

**Issue by shiragoodman** [department-of-veterans-affairs/va.gov-team#55857]
#### VFS team name
VA Online Scheduling (VAOS)

#### VFS product name
VA Online Scheduling

#### VFS feature name
Appointment List

#### Point of Contact/Reviewers
Brian DeConinck - @briandeconinck - Accessibility

*For more information on how to interpret this ticket, please refer to the [Anatomy of a Staging Review issue ticket](https://depo-platform-documentation.scrollhelp.site/collaboration-cycle/Anatomy-of-a-Staging-Review-Issue-ticket.2060320997.html) guidance on Platform Website.

---

### Platform Issue
Design components or patterns don't align with Design System guidelines.

### Issue Details
Each Details link has an `aria-label` that provides a unique accessible name, which is great! However, the `aria-label` isn't being announced consistently by screen readers when a user tabs to that link.

### Link, screenshot or steps to recreate
Steps to reproduce:
1. On a Windows machine, log into staging with a test user and navigate to [Your appointments](https://staging.va.gov/health-care/schedule-view-va-appointments/appointments/).
2. Activate [NVDA](https://www.nvaccess.org/download/).
3. Press [Tab] to move focus to a Details link.
4. Note that the link is announced as "Details" and not as its `aria-label`.
5. On a Mac, log into staging with a test user and navigate to [Your appointments](https://staging.va.gov/health-care/schedule-view-va-appointments/appointments/).
6. Activate VoiceOver.
7. Press [Tab] to move focus to a Details link.
8. Note that the link's `aria-label` is announced as expected.

Identical behavior isn't required across all screen readers, but this is a pretty significant degradation.

### VA.gov Experience Standard
[Category Number 04, Issue Number 07](https://depo-platform-documentation.scrollhelp.site/collaboration-cycle/VA.gov-experience-standards.1683980311.html)

### Other References
WCAG SC 3.2.4 AA

### Platform Recommendation
I believe the issue is with how you're using the `va-link` component. You've added an `aria-label` to the component, but `aria-label` [isn't listed as a supported attribute in Storybook](https://design.va.gov/storybook/?path=/docs/components-va-link--default). Since it's not an attribute supported by the component, the `aria-label` doesn't get passed down to the shadow DOM `a` element.

Different screen readers parse web components differently. Apparently MacOS VoiceOver detects the `aria-label` on the component and announces it, but NVDA doesn't see it in the shadow DOM and doesn't announce it.

Using browser devtools, when I replace the `va-link` component with a standard `a` link, the `aria-label` is announced as expected. It's still probably worth reaching out to the Design System Team to verify that this is a correct diagnosis, however.

Two potential solutions:
- The DST deploys an update to the `va-link` component to handle `aria-label`.
- You update the Details links to just be `a` links without using the web component.

---

### VFS Guidance

- Close the ticket when the issue has been resolved or validated by your Product Owner
- If your team has additional questions or needs Platform help validating the issue, please comment on the ticket
- Some feedback provided may be out of scope for your iteration of the product, however, Platform's OCTO leadership has stated that all identified issues need to be documented and it is still your responsibility to resolve the issue.
- If you do not believe that this Staging Review issue ticket is the responsibility of your team, comment below providing an explanation and who you believe is responsible. Please tag the Point of Contact/Reviewers. Governance team will research and will follow up.

**Comment by outerpress** [comment]
@humancompanion-usds should we file a ticket for the platform to update the va-link component? We might have to revert to using <a> tags in the meantime but I want to make sure that work is captured either way.

**Comment by ldelacosta** [comment]
**Adding a note:** VAOS FE ticket for this work [#56167](https://app.zenhub.com/workspaces/vaos-team-603fdef281af6500110a1691/issues/gh/department-of-veterans-affairs/va.gov-team/56167)

**Comment by ldelacosta** [comment]
@shiragoodman  - We completed this work in #56167. cc: @JRRoof
